### ISSUE TITLE
studio.asyncapi.com only supports version 2.2.0

### ISSUE THREAD

**Issue by DJGummikuh** [asyncapi/studio#252]
#### Describe the bug
I am trying to edit an older async api yaml file that is based on Version 2.0.0 of the specification
this is failing because the editor only allows 2.2.0 to be used.

#### How to Reproduce
Open any file that is not based on the 2.2.0 specification
![image](https://user-images.githubusercontent.com/5188486/151785309-30db2c12-2307-4612-b454-df64b5ad153d.png)

#### Expected behavior
The editor should either support older standards, clearly define that it doesn't and (if that's the route to go) provide hints on what to do in these situations.

**Comment by magicmatatjahu** [comment]
@DJGummikuh Hello! We have bug with that info about unsupported `2.0.0` version but it's still supported, trust me :) 

As I see in the "file preview" on the right side of the editor you have errors in your spec (check that red section at the bottom of your spec). You should click on the terminal section (at the bottom of the editor) to see errors and fix them:

![image](https://user-images.githubusercontent.com/20404945/151786666-67efcf3e-4f25-4bc3-939a-d72a8849cc1a.png)

**Comment by derberg** [comment]
We should definitely not underline `2.0.0` with yellow saying `Value is not accepted` as it is misleading. It already prevented me from digging further for what are the other reasons to not render my document, cause if the version is not supported, why bother

**Comment by fmvilas** [comment]
Also, leaving this comment from @DJGummikuh for the record: https://twitter.com/DJGummikuh/status/1488110968508928000. Looks like two different bugs? 🤔

**Comment by magicmatatjahu** [comment]
@fmvilas If you have broken 2.0.0 spec you cannot make conversion. Converting 2.0.0 -> 2.2.0 is just a version change but still, it is not good to make conversion invalid spec and even if we will make it `2.2.0` will be still invalid.

**Comment by DJGummikuh** [comment]
My issue is that the entire Document is simply red
I have no clue what exactly causes it and the marker at the bottom is also kinda unspecific :D
![image](https://user-images.githubusercontent.com/5188486/151790345-230026bb-a145-4d42-9a0d-8f9b50f9c4c5.png)

**Comment by magicmatatjahu** [comment]
Why does this error (about unsupported 2.0.0) occur? A single instance of the Editor (Monaco editor) can accept a single JSON Scheme for validation. We can say that we have two validations (on the side of the editor itself to show the description of each fields, autocomplete etc) and on the side of our ParserJS. The problem is that the user can change the spec version in the editor at any time and how to check such a change? Of course we can transform raw input to the JSON and check "asyncapi" field but it will worsen the optimization of the whole validation process - for small specs no but for some big ones yes. Therefore, currently the JSON Scheme change for the editor only happens when the spec is valid. This is not a perfect solution but it is a faster solution. The best solution would be to pass the JSON Scheme to the editor with all supported specs, using `oneOf`, but we need to change the paths for `#/definitions` because each spec version has a separate definitions (we can't merge them). I will try handle that in this week.

**Comment by DJGummikuh** [comment]
nah the first spec is only parts, I was copying around to findout what's causing it. The red element was because I had a ref to the lower part of the document (which I didn't copy in). The issue is somewhere in the payload I guess..
I de-corporated (stupid NDAs..) the file and will provide it her:

`asyncapi: '2.0.0'
info:
  title: Some software transmission results API
  version: '1.0.0'
  description: |
    Some software publishes messages containing status information for each individual file that is processed. There is one virtual topic for every job in Some software. The topic names follow the pattern: VirtualTopic.Somesoftware.transmissionresults.<job-id> where 'job-id' is the id of one specific job.
  contact:
    name: "Some software Team"
    email: "mailgroupSome software@Some Company.de"

servers:
  mosquitto:
    url: https://en-Somesoftware-demo.SomeCompany.de:8800
    protocol: JMS
    description: Some software demo system (AcitveMQ Port - 8800 is not exposed yet)

channels:
  VirtualTopic.Somesoftware.transmissionresults.job-id:
    publish:
      description: "On this topic Some software publishes status information for each individual file that is processed by the job with the id 'job-id'."
      message:
        $ref: "#/components/messages/statusMessage"

components:
  messages:
    statusMessage:
        summary: "Data transfer object of the status message"
        description: "JSON deserialized dto containing status information of one processed file."
        payload:
          type: object
          properties:
            status:
              type: string
              description: "The status of the processed file."
              enum:
                - FAILURE
                - SUCCESS
            fileName:
              type: string
              description: "The name of the processed file."
              examples:
                - "file1.txt"
            traceId:
              type: string
              description: "Unique Id for logging purposes."
              examples:
                - "424c2d6c-f04b-441a-add8-bed071f6b27b"
            detectionTime:
              type: long
              description: "The time the file was detected by Some software (format: UNIX Epoch time in ms)."
              examples:
                - 1572346982000
            eventTime:
              type: long
              description: "The time when the message was sent (format: UNIX Epoch time in ms)."
              examples:
                - 1572346982000
            targetLocationId:
              type: string
              description: "The id of the target location the file should be transmitted to."
              examples:
                - "Location 1"
            failureType:
              type: string
              description: "Type of failure. Only available if the status is 'FAILURE'."
              enum:
                - reception-failure
                - internal-failure
                - transmission-failure
            remainingRetries:
              type: integer
              description: "Number of remaining retries. A value of 0 means that there are no more retry attempts and the file is moved to the 'failed' directory of the job. Only available if the status is 'FAILURE'."
              minimum: 0`

**Comment by magicmatatjahu** [comment]
@DJGummikuh Thanks very much! Yeah, it's weird that you have invalid two schemas and whole spec is marked as invalid. I will check that, because probably it's not a problem on the Studio side but on the ParserJS side - studio only consumes the output from ParserJS () and marks that sections which are invalid by info from parser. 

I updated your spec. First of all, you have invalid format of the `info.contact.email` field. You should fix it and also you used `long` type in two schemas. In JSON Schema we have only types: `string`, `boolean`, `number`, `integer` (you can use it for `long`), `array`, `object` and also `null` - others are invalid:

There is a valid spec:

```yaml
asyncapi: '2.0.0'
info:
  title: Some software transmission results API
  version: '1.0.0'
  description: |
    Some software publishes messages containing status information for each individual file that is processed. There is one virtual topic for every job in Some software. The topic names follow the pattern: VirtualTopic.Somesoftware.transmissionresults.<job-id> where 'job-id' is the id of one specific job.
  contact:
    name: "Some software Team"
    email: "mailgroupSomesoftware@somecompany.de"

servers:
  mosquitto:
    url: https://en-Somesoftware-demo.SomeCompany.de:8800
    protocol: JMS
    description: Some software demo system (AcitveMQ Port - 8800 is not exposed yet)

channels:
  VirtualTopic.Somesoftware.transmissionresults.job-id:
    publish:
      description: "On this topic Some software publishes status information for each individual file that is processed by the job with the id 'job-id'."
      message:
        $ref: "#/components/messages/statusMessage"

components:
  messages:
    statusMessage:
        summary: "Data transfer object of the status message"
        description: "JSON deserialized dto containing status information of one processed file."
        payload:
          type: object
          properties:
            status:
              type: string
              description: "The status of the processed file."
              enum:
                - FAILURE
                - SUCCESS
            fileName:
              type: string
              description: "The name of the processed file."
              examples:
                - "file1.txt"
            traceId:
              type: string
              description: "Unique Id for logging purposes."
              examples:
                - "424c2d6c-f04b-441a-add8-bed071f6b27b"
            detectionTime:
              type: integer
              description: "The time the file was detected by Some software (format: UNIX Epoch time in ms)."
              examples:
                - 1572346982000
            eventTime:
              type: integer
              description: "The time when the message was sent (format: UNIX Epoch time in ms)."
              examples:
                - 1572346982000
            targetLocationId:
              type: string
              description: "The id of the target location the file should be transmitted to."
              examples:
                - "Location 1"
            failureType:
              type: string
              description: "Type of failure. Only available if the status is 'FAILURE'."
              enum:
                - reception-failure
                - internal-failure
                - transmission-failure
            remainingRetries:
              type: integer
              description: "Number of remaining retries. A value of 0 means that there are no more retry attempts and the file is moved to the 'failed' directory of the job. Only available if the status is 'FAILURE'."
              minimum: 0
```

**Comment by DJGummikuh** [comment]
We originally built this file using this editor so it must have worked some time in the past.

Yeah the e-mail error was because I Search and replaced our company and product name ^^ the original was a valid e-mail :D
Thanks for cleaning this up though, will give it a thorough look and try to figure out what went wrong!

**Comment by magicmatatjahu** [comment]
@DJGummikuh No problem :) That issue will be closed when we fix that strange behaviour of `unsupported 2.0.0 spec`

**Comment by imabp** [comment]
@DJGummikuh the left one is your version of spec.
I guess, replacing `long` with `integer` worked 
![image](https://user-images.githubusercontent.com/53480076/151804045-6cac4138-bb2c-433d-a70d-1f24ef32ed6d.png)

**Comment by DJGummikuh** [comment]
Yes I can confirm. I used a fresh copy, replaced long with integer and now it's working. However, the object type we use actually IS supposed to be long (ok in the json that we generate it makes no difference, but for java which is parsing it this is relevant). So while it's working now I wonder whether it is ok the way it is :thinking:
